src/psyclone/gocean1p0.py

```diff
@@ -883,6 +883,20 @@ def __init__(self):
         # Get a reference to the namespace manager
         self._name_space_manager = NameSpaceFactory().create()
 
+    def reference_accesses(self, var_accesses):
+        '''Get all variable access information. All accesses are marked
+        according to the kernel metadata.
+
+        :param var_accesses: VariablesAccessInfo instance that stores the\
+            information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+        '''
+        for arg in self.arguments.args:
```

Comment:
Again, you need to check whether the args are scalars or not (I think). It may be possible to do this using the same code as for dynamo0p3.

Reply:
All fixed, but see above.